src/controllers/dynakube/activegate/internal/proxy/reconciler.go

```diff
@@ -56,16 +59,16 @@ func (r *Reconciler) generateForDynakube(ctx context.Context, dynakube *dynatrac
 	}
 
 	coreLabels := kubeobjects.NewCoreLabels(dynakube.Name, kubeobjects.ActiveGateComponentLabel)
-	secret := &corev1.Secret{
-		TypeMeta: metav1.TypeMeta{},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      capability.BuildProxySecretName(),
-			Namespace: r.dynakube.Namespace,
-			Labels:    coreLabels.BuildMatchLabels(),
-		},
-		Data: data,
-		Type: corev1.SecretTypeOpaque,
+	secret, err := kubeobjects.CreateSecret(r.scheme, r.dynakube,
+		kubeobjects.NewSecretTypeModifier(corev1.SecretTypeOpaque),
+		kubeobjects.NewSecretLabelsModifier(coreLabels.BuildMatchLabels()),
+		kubeobjects.NewSecretNameModifier(capability.BuildProxySecretName()),
+		kubeobjects.NewSecretNamespaceModifier(r.dynakube.Namespace),
+		kubeobjects.NewSecretDataModifier(data))
```

Comment:
❤️ 
(nitpick: maybe have an order that would reflect the yaml, like, 1.name 2.namespace 3.labels 4.SecretType 5.Data)